torch/csrc/cuda/memory_snapshot.cpp

```diff
@@ -317,9 +317,53 @@ std::string _memory_snapshot_pickled() {
     traces.push_back(trace);
   }
 
+  auto allocator_settings = new_dict();
+  IValue last_allocator_settings_s = "PYTORCH_CUDA_ALLOC_CONF";
+  IValue max_split_size_s = "max_split_size";
+  IValue garbage_collection_threshold_s = "garbage_collection_threshold";
+  IValue expandable_segments_s = "expandable_segments";
+  IValue pinned_num_register_threads_s = "pinned_num_register_threads";
+#ifdef USE_ROCM
+  IValue release_lock_on_malloc_s = "release_lock_on_hipmalloc";
+  IValue pinned_use_host_register_s = "pinned_use_hip_host_register";
```

Comment:
nit: I wouldn't change the format for ROCM builds because then any tools that consume this might not handle it. I'd just use the work cuda for both.

Reply:
done!